axelor-stock/src/main/java/com/axelor/apps/stock/web/StockMoveController.java

```diff
@@ -459,4 +459,17 @@ public void setAvailableStatus(ActionRequest request, ActionResponse response) {
     stockMoveService.setAvailableStatus(stockMove);
     response.setValue("stockMoveLineList", stockMove.getStockMoveLineList());
   }
+
+  public void updateMoveLineFilterOnAvailableproduct(
+      ActionRequest request, ActionResponse response) {
+    Context context = request.getContext();
+    if (context.containsKey("filterOnAvailableProducts")) {
+      boolean filterOnAvailableProducts = (boolean) context.get("filterOnAvailableProducts");
+      StockMove stockMove = context.asType(StockMove.class);
+      for (StockMoveLine stockMoveLine : stockMove.getStockMoveLineList()) {
```

Comment:
Put null check on getStockMoveLineList(), it will give error when method called with new StockMove and without lines.